controllers/runner_controller.go

```diff
@@ -382,6 +382,9 @@ func (r *RunnerReconciler) newPod(runner v1alpha1.Runner) (corev1.Pod, error) {
 
 	if len(runner.Spec.Containers) != 0 {
 		pod.Spec.Containers = runner.Spec.Containers
+		for i := 0; i < len(pod.Spec.Containers); i++ {
+			pod.Spec.Containers[i].Env = append(pod.Spec.Containers[i].Env, env...)
```

Comment:
I guess this should better be done only when `pod.Spec.Containers[i].Name == containerName`. Otherwise we may end up giving the runner token to unnecessary containers like dind or other user-defined sidecar containers